packages/roosterjs-editor-core/lib/coreApi/selectTable.ts

```diff
@@ -0,0 +1,206 @@
+import {
+    EditorCore,
+    SelectionRangeTypes,
+    TableSelection,
+    SelectTable,
+} from 'roosterjs-editor-types';
+
+const TABLE_ID = 'tableSelected';
+const CONTENT_DIV_ID = 'contentDiv_';
+
+let styleElement: HTMLStyleElement;
+
+/**
+ * @internal
+ * Select a table and save data of the selected range
+ * @param core The EditorCore object
+ * @param table table to select
+ * @param coordinates first and last cell of the selection, if this parameter is null, instead of
+ * selecting, will unselect the table.
+ * @returns true if successful
+ */
+export const selectTable: SelectTable = (
+    core: EditorCore,
+    table: HTMLTableElement,
+    coordinates?: TableSelection
+) => {
+    const doc = core.contentDiv.ownerDocument;
+
+    if (coordinates) {
+        ensureUniqueContentDivId(core);
+        table.id = ensureUniqueId(table, doc);
+        const ranges = select(table, coordinates, core);
+        return {
+            type: SelectionRangeTypes.TableSelection,
+            ranges,
+            table,
+            areAllCollapsed: ranges.filter(range => range?.collapsed).length == ranges.length,
+            coordinates,
+        };
+    }
+
+    unselect();
+
+    if (!coordinates && styleElement) {
+        styleElement.parentElement.removeChild(styleElement);
+        styleElement = null;
+    }
+    return null;
+};
+
+function buildCss(
+    table: HTMLTableElement,
+    coordinates: TableSelection,
+    contentDivSelector: string
+): { css: string; ranges: Range[] } {
+    coordinates = normalizeTableSelection(coordinates);
+    const tr1 = coordinates.firstCell.y + 1;
+    const td1 = coordinates.firstCell.x + 1;
+    const tr2 = coordinates.lastCell.y + 1;
+    const td2 = coordinates.lastCell.x + 1;
+    const ranges: Range[] = [];
+    const tbodySelector = table.querySelector('tbody') ? '>tbody' : '';
```

Comment:
From my test, no matter if there is tbody, the ">tbody" in the selector always works.

PS, also need to consider THead and TFoot scenario

Reply:
and better not use querySelector here since if there is nested table inside, it will also be selected

Reply:
Added some checks to see whether the table contains a tbody, thead and tfoot from line 70 to 94